src/tests/nativeaot/SmokeTests/SharedLibrary/SharedLibraryDriver.mjs

```diff
@@ -0,0 +1,39 @@
+import { readFile } from 'node:fs/promises';
+import { WASI } from 'wasi';
+import { argv, env } from 'node:process';
+
+const wasi = new WASI({
+  version: 'preview1',
+  args: argv,
+  env
+});
+
+const wasm = await WebAssembly.compile(  
+  await readFile(new URL("./SharedLibrary.wasm", import.meta.url)),
+);
+
+const instance = await WebAssembly.instantiate(wasm, wasi.getImportObject());
+
+wasi.initialize(instance);
+
+if (instance.exports.ReturnsPrimitiveInt() != 10)
+    process.exit(1);
+
+if (instance.exports.ReturnsPrimitiveBool() != 1)
+    process.exit(2);
+
+if (instance.exports.ReturnsPrimitiveChar() != 97) // 'a'
+    process.exit(3);
+
+// As long as no unmanaged exception is thrown managed class loaders were initialized successfully.
+instance.exports.EnsureManagedClassLoaders();
+
+// #if !CODEGEN_WASI - for some reason tries to create a background thread?
+// if (instance.exports.CheckSimpleGCCollect() != 100)
+//     process.exit(4);
+
+// #if !CODEGEN_WASI - enable when we support exception handling
```

Comment:
I actually tried some simple exception handling examples and somehow they worked, but in some other scenarios it didn't. I'm not sure what it depends on - LLVM optimisations?

Reply:
For example, I had something like this in my sample and it compiled and ran fine:

```cs
	[UnmanagedCallersOnly(EntryPoint = "hello")]
	public static void Hello() {
		try {
			Console.WriteLine("Hello from exception");
		} catch (Exception e) {
			Console.WriteLine(e);
		}
		Console.WriteLine("Hello from C#");
		MyMethod("lores ipsum");
	}
```

Reply:
What did what did not work?

There is a dependency on `emscripten_get_callstack` inside exception handling (it's called on each `throw`), so I would _expect_ things to fail relatively deterministically.

Reply:
> Edit: that example doesn't `throw`, so I suppose it is expected to work.

Oh my god, this is ridiculous. I need sleep 😬